[Bug]: Generated sdk test script fails

### Package name

workspace

### Package version

13.0.0-next

### Reproduction steps

Create a project using `npm create @o3r my-project`.
Generate an sdk with `npm exec ng g sdk my-sdk`.
Run test in the sdk with `npm run test` 

### Current result

```
Error: Cannot find module '../../jest.config.ut'
Require stack:
- C:\git_clones\test-it\libs\pouet\jest.config.js
- C:\git_clones\test-it\node_modules\jest-util\build\requireOrImportModule.js
- C:\git_clones\test-it\node_modules\jest-util\build\index.js
- C:\git_clones\test-it\node_modules\@jest\core\build\FailedTestsInteractiveMode.js
- C:\git_clones\test-it\node_modules\@jest\core\build\plugins\FailedTestsInteractive.js
- C:\git_clones\test-it\node_modules\@jest\core\build\watch.js
- C:\git_clones\test-it\node_modules\@jest\core\build\cli\index.js
- C:\git_clones\test-it\node_modules\@jest\core\build\index.js
- C:\git_clones\test-it\node_modules\jest-cli\build\run.js
- C:\git_clones\test-it\node_modules\jest-cli\build\index.js
- C:\git_clones\test-it\node_modules\jest-cli\bin\jest.js
- C:\git_clones\test-it\node_modules\jest\bin\jest.js
    at Function._resolveFilename (node:internal/modules/cjs/loader:1401:15)
    at defaultResolveImpl (node:internal/modules/cjs/loader:1057:19)
    at resolveForCJSWithHooks (node:internal/modules/cjs/loader:1062:22)
    at Function._load (node:internal/modules/cjs/loader:1211:37)
    at TracingChannel.traceSync (node:diagnostics_channel:322:14)
    at wrapModuleLoad (node:internal/modules/cjs/loader:235:24)
    at Module.require (node:internal/modules/cjs/loader:1487:12)
    at require (node:internal/modules/helpers:135:16)
    at Object.<anonymous> (C:\git_clones\test-it\libs\pouet\jest.config.js:1:30)
    at Module._compile (node:internal/modules/cjs/loader:1730:14)
```

### Expected result

Tests run without errors 

### Additional comments

_No response_

--- a/packages/@o3r/testing/schematics/ng-add/index.ts
+++ b/packages/@o3r/testing/schematics/ng-add/index.ts
@@ -57,7 +57,6 @@ const devDependenciesToInstall = [
   'pixelmatch',
   'pngjs',
   '@angular-devkit/build-angular',
-  '@angular/animations',
   '@angular/core',
   '@angular/common'
 ];

--- a/packages/@o3r/workspace/schematics/index.it.spec.ts
+++ b/packages/@o3r/workspace/schematics/index.it.spec.ts
@@ -39,6 +39,7 @@ describe('new otter workspace', () => {
 
     expect(() => packageManagerInstall(execAppOptions)).not.toThrow();
     expect(() => packageManagerRunOnProject('@my-sdk/sdk', isInWorkspace, { script: 'build' }, execAppOptions)).not.toThrow();
+    expect(() => packageManagerRunOnProject('@my-sdk/sdk', isInWorkspace, { script: 'test' }, execAppOptions)).not.toThrow();
   });
 
   test('should add sdk to an existing workspace with local spec', async () => {

--- a/packages/@o3r/workspace/schematics/sdk/index.ts
+++ b/packages/@o3r/workspace/schematics/sdk/index.ts
@@ -10,6 +10,7 @@ import {
   MergeStrategy,
   mergeWith,
   move,
+  noop,
   renameTemplateFiles,
   Rule,
   strings,
@@ -96,14 +97,16 @@ function generateSdkFn(options: NgGenerateSdkSchema): Rule {
     /** Path to the folder where generate the new SDK */
     const targetPath = path.posix.join(options.path || defaultRoot, cleanName);
 
-    const addModuleSpecificFiles = () => mergeWith(apply(url('./templates'), [
-      template({
-        ...options,
-        rootRelativePath: path.posix.relative(targetPath, tree.root.path.replace(/^\//, './'))
-      }),
-      move(targetPath),
-      renameTemplateFiles()
-    ]), MergeStrategy.Overwrite);
+    const overrideFilesIfRootJestConfigUTExists = tree.exists('jest.config.ut.js')
+      ? mergeWith(apply(url('./templates'), [
+        template({
+          ...options,
+          rootRelativePath: path.posix.relative(targetPath, tree.root.path.replace(/^\//, './'))
+        }),
+        move(targetPath),
+        renameTemplateFiles()
+      ]), MergeStrategy.Overwrite)
+      : noop();
 
     const packageManager = getPackageManager({ workspaceConfig });
     const specExtension = options.specPackagePath ? path.extname(options.specPackagePath) : '.yaml';
@@ -151,7 +154,7 @@ function generateSdkFn(options: NgGenerateSdkSchema): Rule {
       isNx ? nxRegisterProjectTasks(options, targetPath, cleanName) : ngRegisterProjectTasks(options, targetPath, cleanName),
       updateTsConfig(targetPath, projectName, scope),
       cleanStandaloneFiles(targetPath),
-      addModuleSpecificFiles(),
+      overrideFilesIfRootJestConfigUTExists,
       setupDependencies({
         dependencies,
         skipInstall: options.skipInstall,
@@ -159,7 +162,7 @@ function generateSdkFn(options: NgGenerateSdkSchema): Rule {
         projectName: cleanName,
         runAfterTasks: sdkGenerationTasks
       })
-    ])(tree, context);
+    ]);
   };
 }
 

